### PR TITLE
solution added

### PR DESCRIPTION
**Pull request by Vosker**

### CODE REVIEW

**Review by matviichuk92 — APPROVED** [review]
good job!

**Review by KostiantynVorobiov — APPROVED** [review]
I think it's ok

File: src/main/java/com/internet/shop/Application.java [hunk 1]
```diff
@@ -2,32 +2,95 @@
...
+
+        System.out.println("Users after update");
+        String usersName = "Kostya";
+        User user = new User(usersName, user2.getLogin(), user2.getPassword());
+        user.setId(2L);
+        userService.update(user);
+        System.out.println(userService.getAll().toString());
+
+        System.out.println("Users after remove");
+        userService.delete(3L);
```
Comment (tkhambir) [inline src/main/java/com/internet/shop/Application.java:55]: We should create ShoppingCart for every User and delete it if we delete user.

File: src/main/java/com/internet/shop/dao/impl/OrderDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    @Override
+    public List<Order> getAll() {
+        return Storage.orders;
+    }
+
+    @Override
+    public Order update(Order order) {
+        IntStream.range(0, Storage.orders.size())
+                .filter(i -> Storage.orders.get(i).getId().equals(order.getId()))
+                .forEach(item -> Storage.orders.set(item, order));
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/impl/OrderDaoImpl.java]:
```suggestion
                .forEach(i -> Storage.orders.set(i, order));
```
it's index, not item

File: src/main/java/com/internet/shop/dao/impl/ShoppingCartDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    @Override
+    public List<ShoppingCart> getAll() {
+        return Storage.shoppingCarts;
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart shoppingCart) {
+        IntStream.range(0, Storage.shoppingCarts.size())
+                .filter(i -> Storage.shoppingCarts.get(i).getId().equals(shoppingCart.getId()))
+                .forEach(item -> Storage.shoppingCarts.set(item, shoppingCart));
```
Comment (RomanDubovskyi) [inline src/main/java/com/internet/shop/dao/impl/ShoppingCartDaoImpl.java]: same here and everywhere

**Review by RomanDubovskyi — COMMENTED** [review]
gj, left minor naming comment

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**